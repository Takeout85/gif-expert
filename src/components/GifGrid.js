import React from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import Loading from "./Loading";
import GifGridItem from "./GifGridItem";
import PropTypes from "prop-types";

const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      {loading && <Loading />}

      <li className="category">
        <h3>{category.toUpperCase()}</h3>
        <div className="card-grid">
          {images.map((img) => (
            <GifGridItem key={img.id} {...img} />
          ))}
        </div>
        <hr />
      </li>
    </>
  );
};

GifGrid.prototype = {
  category: PropTypes.string.isRequired,
};

export default GifGrid;
