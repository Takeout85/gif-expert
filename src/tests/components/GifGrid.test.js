import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import GifGrid from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe("Tests on the <GifGrid /> component", () => {
  const category = "Kurisu";

  test("should show the component correctly", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });

    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("should show objects when images are loaded useFetchGifs", () => {
    const gifs = [
      {
        id: "ABC",
        url: "https://localhost/anything.jpg",
        title: "Cualquier cosa",
      },
      {
        id: "DEF",
        url: "https://localhost/anything2.jpg",
        title: "Cualquier cosa2",
      },
    ];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("p").exists()).toBe(false);
    expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
  });
});
