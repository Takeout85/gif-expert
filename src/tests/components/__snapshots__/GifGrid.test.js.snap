// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on the <GifGrid /> component should show objects when images are loaded useFetchGifs 1`] = `
<Fragment>
  <li
    className="category"
  >
    <h3>
      KURISU
    </h3>
    <div
      className="card-grid"
    >
      <GifGridItem
        id="ABC"
        key="ABC"
        title="Cualquier cosa"
        url="https://localhost/anything.jpg"
      />
      <GifGridItem
        id="DEF"
        key="DEF"
        title="Cualquier cosa2"
        url="https://localhost/anything2.jpg"
      />
    </div>
    <hr />
  </li>
</Fragment>
`;

exports[`Tests on the <GifGrid /> component should show the component correctly 1`] = `
<Fragment>
  <Loading />
  <li
    className="category"
  >
    <h3>
      KURISU
    </h3>
    <div
      className="card-grid"
    />
    <hr />
  </li>
</Fragment>
`;
